### ISSUE TITLE
False positives when 'type' builtin is overwritten

### ISSUE THREAD

**Issue by mharding-hpe** [PyCQA/pylint#10994]
### Bug description

```python
def convert_type(x, type): # pylint: disable=missing-function-docstring,redefined-builtin
    return type(x)

convert_type(12.34, str).split('.')
```

For the final line, `pylint` reports:
```
E1101: Class 'float' has no 'split' member (no-member)
```

If `type` is renamed in the function definition, then `pylint` reports no errors.

The exact false positive reported is not always the same. I believe the fundamental problem is that `pylint` does not know how to handle when `type` has been shadowed, and acts as though it has not been. Hence in the above example, it thinks that the function call returned the `float` class, since that is what it would be if the built-in `type` had actually been used.

I would like to mention that I am not an advocate for shadowing builtins like this. I came across this issue when I was working on existing code authored by someone else. But even though shadowing builtins is bad practice, this does seem like something `pylint` should be able to understand.

### Configuration

```ini

```

### Command used

```shell
pylint --errors-only a.py
```

### Pylint output

```
************* Module a
/tmp/a.py:4:0: E1101: Class 'float' has no 'split' member (no-member)
```

### Expected behavior

I would expect it to report no errors, like it does when you write the same code but without redefining `type`.

### Pylint version

```shell
pylint 4.0.5
astroid 4.0.4
Python 3.13.3 (main, May  9 2025, 23:49:05) [GCC 12.2.0]
```

### OS / Environment

I discovered this in the Debian Python 3:13 docker image. It seems to be Debian 12 (bookworm).

### Additional dependencies

```python

```

**Comment by MCharming98** [comment]
I took a look at this and believe the cause is that the `astroid` brain transformer for the type builtin triggers based solely on the identifier name, ignoring local shadowing (LEGB rules).

Evidence:
- In `astroid/brain/brain_builtin_inference.py`, the `infer_type` registration uses a predicate that only checks `node.func.name`. It doesn’t currently use `lookup()` or `infer()` to verify if the name actually refers to the builtin.
```python
    if isinstance(node.func, nodes.Name):
        return node.func.name == builtin_name
```

- Interestingly, `brain_type.py` (for subscripts) handles this correctly by using `node.scope().lookup("type")` to check for shadowing.
```python
    node_scope, _ = node.scope().lookup("type")
    if not (isinstance(node_scope, nodes.Module) and node_scope.qname() == "builtins"):
        raise UseInferenceDefault()
```

Suggested Fix:
- Use `node.scope().lookup(builtin_name)` to verify the definition origin, similar to what is done in `astroid/brain/brain_type.py`.
- Alternatively, the individual inference functions could raise UseInferenceDefault if shadowing is detected via node.scope().lookup().

<details>
<summary>Note on this analysis</summary>

I’m currently developing Repellent AI, a specialized Root Cause Analysis (RCA) agent designed to reduce maintainer triage time. It recently achieved 88% accuracy on the SWE-bench Verified dataset.
 
I am testing on live issues like this one to validate the quality of the analysis. My goal is to provide helpful, actionable signals rather than automated noise. I'd love to hear if this level of detail is useful for you!

</details>
